tools/log-collector/log_collector.go

```diff
@@ -474,258 +321,186 @@ func (l *LogCollector) zipDir() error {
-					if eLrr != nil {
-						return nil, eLrr
-					}
-				}
-				eLrr := l.writeYaml(resourceDir, obj)
-				if eLrr != nil {
-					return nil, eLrr
-				}
+			var resObjects unstructured.UnstructuredList
+			resObject, err := l.filterResourceObjects(getAPIGroupVersionResourcePath(groupVersion), &resources[index])
+			if err != nil {
+				return err
 			}
-		}
-	}
-	return resourceMap, nil
-}
+			resObjects.Items = append(resObjects.Items, resObject.Items...)
 
-// admissionRegistrationGroup gets all the resources related admissionRegistration and writes their YAML
-func (l *LogCollector) admissionRegistrationGroup(apiGroups []*apiv1.APIGroup) error {
-	log.Info("Checking Admission Registration Group")
-	admissionGV := getGVByGroup(apiGroups, AdmissionRegistrationGroup, true)
-	if len(admissionGV) != 0 {
-		admissionGVResources, agErr := l.getAPIGVResources(admissionGV[0])
-		if agErr != nil {
-			return agErr
-		}
-		for index := range admissionGVResources {
-			objectList := l.getResourceObjects(getAPIGroupVersionResourcePath(admissionGV[0]), &admissionGVResources[index])
-			resourceDir := filepath.Join(admissionGVResources[index].Kind)
-
-			for _, obj := range objectList.Items {
-				if strings.HasPrefix(obj.GetName(), "k8s-triliovault") {
-					eLrr := l.writeYaml(resourceDir, obj)
-					if eLrr != nil {
-						return eLrr
-					}
+			if l.CheckIsOpenshift() {
+				ocpObj, oErr := l.getResourceObjectsWithOwnerRef(getAPIGroupVersionResourcePath(groupVersion), &resources[index])
+				if oErr != nil {
+					return oErr
 				}
+				resObjects.Items = append(resObjects.Items, ocpObj.Items...)
 			}
-		}
-	}
-	return nil
-}
 
-// snapshotStorageGroup gets all the resources related snapshot storage and writes their YAML
-func (l *LogCollector) snapshotStorageGroup(apiGroups []*apiv1.APIGroup) error {
-	log.Info("Checking Snapshot Storage Group")
-	snapGV := getGVByGroup(apiGroups, SnapshotStorageGroup, true)
-	if snapGV[0] != "" {
-		snapGVResources, err := l.getAPIGVResources(snapGV[0])
-		if err != nil {
-			return err
-		}
-		volSnapResource := getResourceByName(snapGVResources, VolumeSnapshot)
-		volSnapObjects := l.getResourceObjects(getAPIGroupVersionResourcePath(snapGV[0]), &volSnapResource)
-		for _, obj := range volSnapObjects.Items {
-			resourceDir := filepath.Join(obj.GetKind())
-			eLrr := l.writeYaml(resourceDir, obj)
-			if eLrr != nil {
-				return eLrr
+			resourceMap, err := l.writeObjectsAndLogs(resObjects, resources[index].Kind)
+			if err != nil {
+				return err
 			}
-		}
 
-		volSnapClassResource := getResourceByName(snapGVResources, VolumeSnapshotClass)
-		volSnapClassObjects := l.getResourceObjects(getAPIGroupVersionResourcePath(snapGV[0]), &volSnapClassResource)
-		for _, obj := range volSnapClassObjects.Items {
-			resourceDir := filepath.Join(obj.GetKind())
-			eLrr := l.writeYaml(resourceDir, obj)
-			if eLrr != nil {
-				return eLrr
+			for kind, NsName := range resourceMap {
+				resourceMapList[kind] = NsName
```

Comment:
what is use of this for loop and `resourceMapList`? resourceMapList and resourceMap are same.

Reply:
resourceMap is for particular resource and resourceMapList is for all resources